pkg/workloads/cortex/lib/model/cron.py

```diff
@@ -0,0 +1,1776 @@
+            api_spec: Identical copy of pkg.type.spec.api.API.
+            address: An address with the "host:port" format to where TFS is located.
+            addresses: A list of addresses with the "host:port" format to where the TFS servers are located.
+            tfs_model_dir: Path to where the models are stored within the TFS container.
+            download_dir: Path to where the models are stored.
+            temp_dir: Directory where models are temporarily stored.
+            lock_dir: Directory in which model timestamps are stored.
+        """
+
+        if address and addresses:
+            raise ValueError("address and addresses arguments cannot be passed in at the same time")
+        if not address and not addresses:
+            raise ValueError("must pass in at least one of the two arguments: address or addresses")
+
+        mp.Process.__init__(self, daemon=True)
+
+        self._interval = interval
+        self._api_spec = api_spec
+        self._tfs_model_dir = tfs_model_dir
+        self._download_dir = download_dir
+        self._temp_dir = temp_dir
+        self._lock_dir = lock_dir
+
+        if address:
+            self._tfs_address = address
+            self._tfs_addresses = None
+        else:
+            self._tfs_address = None
+            self._tfs_addresses = addresses
+
+        self._s3_paths = []
+        self._spec_models = CuratedModelResources(self._api_spec["curated_model_resources"])
+        self._local_model_names = self._spec_models.get_local_model_names()
+        self._s3_model_names = self._spec_models.get_s3_model_names()
+        for model_name in self._s3_model_names:
+            if not self._spec_models.is_local(model_name):
+                self._s3_paths.append(self._spec_models[model_name]["model_path"])
+
+        if (
+            self._api_spec["predictor"]["model_path"] is None
+            and self._api_spec["predictor"]["models"] is not None
+            and self._api_spec["predictor"]["models"]["dir"] is not None
+        ):
+            self._is_dir_used = True
+            self._models_dir = self._api_spec["predictor"]["models"]["dir"]
+        else:
+            self._is_dir_used = False
+            self._models_dir = None
+
+        if self._api_spec["predictor"]["type"] == "tensorflow":
+            if self._api_spec["compute"]["inf"] > 0:
+                self._predictor_type = TensorFlowNeuronPredictorType
+            else:
+                self._predictor_type = TensorFlowPredictorType
+        else:
+            raise CortexException(
+                "'tensorflow' predictor type is the only allowed type for this cron"
+            )
+
+        self._ran_once = mp.Event()
+        self._event_stopper = mp.Event()
+        self._stopped = mp.Event()
+
+    def run(self):
+        """
+        mp.Process-specific method.
+        """
+
+        self._old_ts_state = {}
+
+        if self._tfs_address:
```

Comment:
I wouldn't recommend this (and I think it doesn't actually work), because the `self._client` object is literally copied from one process to another, and that may have unintended consequences.